File: syscall/hello_fork.c
#include <stdio.h>
#include <unistd.h>
#include <sys/wait.h>

int main(int argc, char *argv[])
{
    int count = 0;
    printf("Before fork count: %d, pid: %d\n", count, getpid());

    if (fork())
    {
        count += 2;
        printf("Parent After fork count: %d, pid: %d\n", count, getpid());
    }
    else
    {
        count += 3;
        printf("Child After fork count: %d, pid: %d\n", count, getpid());
    }
    return 0;
}

File: syscall/hello_fork.py
import os
import sys

count = 0

print('Before fork count: {}, pid: {}.'.format(count, os.getpid()))
count += 1

child_pid = os.fork()
if child_pid == 0:
    count += 5
    print("Child After fork count: {}, pid: {}.".format(count, os.getpid()))
    sys.exit(os.EX_OK)
else:
    count += 3
    print("Parent After fork count: {}, pid: {}.".format(count, os.getpid()))
